aiida/tools/visualization/graph.py

```diff
@@ -0,0 +1,740 @@
+                          print_func=None):
+        """add nodes and edges from an origin recursively,
+        following incoming links
+
+        :param origin: node or node pk/uuid
+        :type origin: aiida.orm.nodes.node.Node or int
+        :param depth: if not None, stop after travelling a certain depth into the graph (Default value = None)
+        :type depth: None or int
+        :param link_types: filter by subset of link types (Default value = ())
+        :type link_types: tuple or str
+        :param annotate_links: label edges with the link 'label', 'type' or 'both' (Default value = False)
+        :type annotate_links: bool
+        :param origin_style: node style map for origin node (Default value = ())
+        :type origin_style: dict or tuple
+        :param include_process_outputs:  include outgoing links for all processes (Default value = False)
+        :type include_process_outputs: bool
+        :param print_func: a function to stream information to, i.e. print_func(str)
+
+        """
+        # pylint: disable=too-many-arguments
+        origin_node = self._load_node(origin)
+
+        self.add_node(origin_node, style_override=dict(origin_style))
+
+        last_nodes = [origin_node]
+        traversed_pks = [origin_node.pk]
+        cur_depth = 0
+        while last_nodes:
+            cur_depth += 1
+            # checking of maximum descendant depth is set and applies.
+            if depth is not None and cur_depth > depth:
+                break
+            if print_func:
+                print_func("- Depth: {}".format(cur_depth))
+            new_nodes = []
+            for node in last_nodes:
+                incoming_nodes = self.add_incoming(
+                    node, link_types=link_types, annotate_links=annotate_links, return_pks=False)
+                if incoming_nodes and print_func:
+                    print_func("  {} -> {}".format(node.pk, [n.pk for n in incoming_nodes]))
+                new_nodes.extend(incoming_nodes)
+
+                if include_process_outputs and isinstance(node, BaseFactory("aiida.node", "process")):
+                    self.add_outgoing(node, link_types=link_types, annotate_links=annotate_links)
+
+            # ensure the same path isn't traversed multiple times
+            last_nodes = []
+            for new_node in new_nodes:
+                if new_node.pk in traversed_pks:
+                    continue
+                last_nodes.append(new_node)
+                traversed_pks.append(new_node.pk)
+
+    def add_origin_to_target(self,
+                             origin,
+                             target_cls,
+                             target_filters=None,
+                             include_target_inputs=False,
+                             include_target_outputs=False,
+                             origin_style=(),
+                             annotate_links=False):
+        """add nodes and edges from an origin node to target nodes
```

Comment:
If I understand correctly, it's actually rather
```suggestion
        """Add nodes and edges from an origin node to all nodes of a target node class.
```

Reply:
In this case, the function should potentially be renamed `add_origin_to_targets` (also to be consistent with the next function)